pandas/core/dtypes/cast.py

```diff
@@ -1358,6 +1359,24 @@ def find_result_type(left_dtype: DtypeObj, right: Any) -> DtypeObj:
         dtype, _ = infer_dtype_from(right)
         new_dtype = find_common_type([left_dtype, dtype])
 
+        # GH#61671: datetime64[ns] inferred but the value may be out of bounds
+        if (
+            is_datetime64_dtype(new_dtype)
+            and lib.is_scalar(right)
+            and isinstance(right, (np.datetime64, dt.datetime))
+        ):
+            try:
+                ts = Timestamp(right)
+                casted = ts.as_unit("ns")
```

Comment:
i think you can use round_ok=False here to avoid the need for the check on the next line